packages/xo-server/src/api/disk.js

```diff
@@ -165,3 +168,72 @@ resize.params = {
 resize.resolve = {
   vdi: ['id', ['VDI', 'VDI-snapshot'], 'administrate'],
 }
+
+async function handleImport(req, res, { type, name, vmdkData, srId, xapi }) {
+  req.setTimeout(43200000) // 12 hours
+  try {
+    req.length = req.headers['content-length']
+    let vhdStream, size
+    if (type === 'vmdk') {
+      vhdStream = await convertFromVMDK(req, vmdkData.blocksTable)
+      size = vmdkData.size
+    }
+    if (type === 'vhd') {
+      vhdStream = req
+      const footer = await peekFooterFromVhdStream(req)
+      size = footer.currentSize
+    }
+    if (vhdStream === undefined) {
+      throw new Error(
+        `Unknown disk type, expected "vhd" or "vmdk", got ${type}`
+      )
+    }
+    const vdi = await xapi.createVdi({
+      name_label: name,
+      size,
+      sr: srId && this.getObject(srId).$ref,
+    })
+    await xapi.importVdiContent(vdi, vhdStream, VDI_FORMAT_VHD)
+    res.end(format.response(0, vdi.$id))
+  } catch (e) {
+    res.writeHead(500)
+    res.end(format.error(0, new Error(e.message)))
+  }
+}
+
+// type is 'vhd' or 'vmdk'
+async function importDisk({ sr, type, name, vmdkData }) {
+  return {
+    $sendTo: await this.registerHttpRequest(handleImport, {
+      type,
+      name,
+      vmdkData,
+      srId: sr._xapiId,
+      xapi: this.getXapi(sr),
```

Comment:
Why storing the `xapi` object and not the `sr`?

Reply:
?

Reply:
Still no answer on this.

Reply:
sorry, missed it. will do.

Reply:
It was not a request, just a question :slightly_smiling_face:

Reply:
It's a copy/paste of the code in the OVA import, I did not write the original.

I did a copy/paste in the hope that it would go through faster, but that did not happen.

Reply:
Ok :stuck_out_tongue: